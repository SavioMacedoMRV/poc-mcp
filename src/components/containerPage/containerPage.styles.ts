import styled from 'styled-components'
import { cores } from 'resources/cores'
type PropsExtra = {
  $corFundo?: string
  $semCabecalho?: boolean
}

export const Container = styled.div<PropsExtra>`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  background-color: ${({ $corFundo }) => $corFundo || cores.background01};
`

export const Content = styled.div<PropsExtra>`
  display: flex;
  flex-direction: column;
  min-height: ${({$semCabecalho}) =>
    `calc(100cqh - ${$semCabecalho ? 0 : 64}px)`};
`
