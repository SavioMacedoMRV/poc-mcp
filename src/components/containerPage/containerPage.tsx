import React, {ReactNode} from 'react'
import * as S from './containerPage.styles'
import {Cabecalho, CabecalhoProps} from 'components/cabecalho'
import {BotaoScrollTop} from 'components/botaoScrollTop'

interface Props {
  children: ReactNode
  $corFundo?: string
  semCabecalho?: boolean
  comBotaoScrollTop?: boolean
  cabecalhoProps?: CabecalhoProps
}

export const ContainerPage = ({
  comBotaoScrollTop,
  semCabecalho,
  $corFundo,
  children,
  cabecalhoProps,
}: Readonly<Props>) => {
  return (
    <S.Container data-testid={'container-page-ordens'} $corFundo={$corFundo}>
      {!semCabecalho && <Cabecalho {...cabecalhoProps} />}
      <S.Content
        data-testid={'content-page-ordens'}
        $semCabecalho={semCabecalho}
      >
        {children}
        {comBotaoScrollTop && <BotaoScrollTop />}
      </S.Content>
    </S.Container>
  )
}
