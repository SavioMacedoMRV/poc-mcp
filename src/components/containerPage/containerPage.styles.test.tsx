import {render} from '@testing-library/react'
import * as S from './containerPage.styles'

describe('ContainerPage.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Container />)
    expect(container).toMatchSnapshot()
  })

  test('Container DEVE ser igual ao snapshot QUANDO for passado uma cor de fundo', () => {
    const {container} = render(<S.Container $corFundo="#ff0000" />)
    expect(container).toMatchSnapshot()
  })

  test('Content DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Content />)
    expect(container).toMatchSnapshot()
  })
})
