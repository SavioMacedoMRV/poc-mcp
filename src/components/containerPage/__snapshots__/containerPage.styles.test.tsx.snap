// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerPage.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  background: #EDF2F0;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ContainerPage.styles Container DEVE ser igual ao snapshot QUANDO for passado uma cor de fundo 1`] = `
.c0 {
  background: #ff0000;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ContainerPage.styles Content DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  min-height: calc(100cqh - 64px);
}

<div>
  <div
    class="c0"
  />
</div>
`;
